components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java

```diff
@@ -2000,4 +2017,120 @@ public static SAMLSSOServiceProviderDO getSPConfig(String tenantDomain, String i
             PrivilegedCarbonContext.endTenantFlow();
         }
     }
+
+    public static LogoutRequest buildLogoutRequest(SAMLSSOServiceProviderDO serviceProviderDO, String subject,
+                                                   String sessionId) throws IdentityException {
+
+        SingleLogoutRequestDTO logoutReqDTO = new SingleLogoutRequestDTO();
+
+        if (StringUtils.isNotBlank(serviceProviderDO.getSloRequestURL())) {
```

Comment:
We can add debug logs saying how we resolve SLO url here.. It will be helpful when tracing issues.